### ISSUE TITLE
Feature request: expose time conversions

### ISSUE THREAD

**Issue by gaffarelj** [tudat-team/tudatpy#28]
A new utility should be added to allow to convert a float or an array of Julian Dates used by Tudat (seconds since J2000), to "normal" Roman dates.

**Comment by gaffarelj** [comment]
To do this, the time conversions included in tudat have to be exposed. These can be seen [here](https://github.com/tudat-team/tudat/blob/834207f11989212d2c25231a8e91a5f0451ad463/include/tudat/astro/basic_astro/timeConversions.h).

**Comment by gaffarelj** [comment]
The documentation is now complete too.

The new module is `tudatpy.astro.time_conversion`, and its associated documentation can be found [here](https://tudatpy.readthedocs.io/en/latest/time_conversion.html).

@FilippoOggionni and @DominicDirkx, could you give the documentation page a quick look, and confirm that there are no issues when using this new module? If so, I will then close this issue.

**Comment by FilippoOggionni** [comment]
Great job on this @gaffarelj ! 

I quickly skimmed through the documentation and everything seems in order. I see that you have not built the conda package yet, correct?

A few remarks from my side:

- since this module relies a lot on time systems and references, it would be nice to have a reference (I think one for the entire page will suffice, it is not needed to add a reference for each entry) - maybe @DominicDirkx can provide a good one?
- since most (if not all) the functions are conversions, in the code snippets it would be nice to have the result of each conversion printed and actually included in the example code. I will do this, so I can also test the module in parallel
- for the reference dates (e.g., 1st January 2000 in Julian days), it would be nice to add this number (and similar ones) to tudat constants, if it is not there yet

**Comment by gaffarelj** [comment]
Indeed, I haven't built the conda package. I actually haven't looked in how to do that yet.

About your remarks:
- [This reference](https://www.ucolick.org/~sla/leapsecs/timescales.html) seems to be very complete, based on the International Astronomical Union resolutions from 1976, and is actually referred to by different softwares.
- Thanks for planning on improving the code snippets this way.
- Most (if not all) of the time constants refered to are taken from the physical constants. I don't think these constants are described anywhere in the API docs. Should I open an issue about this?

**Comment by gaffarelj** [comment]
Actually, these constants are defined in [timeConversions.h](https://github.com/tudat-team/tudat/blob/b8149dc3cc40c14e242599bc0951d7fba6caf3bd/include/tudat/astro/basic_astro/timeConversions.h), and these are the ones set as default inputs (and not hard-coded numbers), see [here for instance](https://github.com/tudat-team/tudatpy/blob/28305a1fe9fbfff576e29c24c5a91e7cda7e6ca6/tudatpy/kernel/expose_astro/expose_time_conversion.cpp#L96).

I then plan on exposing both the [JULIAN_DAY_ON_J2000](https://github.com/tudat-team/tudat/blob/b8149dc3cc40c14e242599bc0951d7fba6caf3bd/include/tudat/astro/basic_astro/timeConversions.h#L44) and [JULIAN_DAY_AT_0_MJD](https://github.com/tudat-team/tudat/blob/b8149dc3cc40c14e242599bc0951d7fba6caf3bd/include/tudat/astro/basic_astro/timeConversions.h#L58) variables as enums in the `time_conversion` API docs.

**Comment by FilippoOggionni** [comment]
> These are actually defined in timeConversions.h, and these are the ones set as default inputs (and not hard-coded numbers), see here for instance.

I see, but I originally meant in the docstrings, where they are hard-coded (see for instance [here](https://github.com/tudat-team/tudat-multidoc/blob/941068d5bc2363c35bf527f3e751c17bdc989f2c/docstrings/astro/time_conversion.yaml#L62)). There, I would rather have something like:
```
- name: epoch_since_julian_day_zero
   type: float, default = constants.JULIAN_DAY_ON_J2000
```
instead of the actual number (the actual number can be made explicit in the description of the function, if needed).

> I then plan on exposing both the JULIAN_DAY_ON_J2000 and JULIAN_DAY_AT_0_MJD variables as enums in the time_conversion API docs.

Regarding this, I don't see the reason to expose again those variables in an ``enum``, as they are already exposed in the ``tudatpy.constants`` module (see the exposition file [here](https://github.com/tudat-team/tudatpy/blob/1b0c8075358ce7b643de8f4930b1b92e0234ffe8/tudatpy/kernel/expose_constants.cpp#L34)).

**Comment by FilippoOggionni** [comment]
I made my changes in 6edd2f6abeabfaf9723e9b6463448b0acc807e8d. I am testing locally and then push it online. 
@gaffarelj I saw that the last few functions do not have examples, is this intended?

**Comment by gaffarelj** [comment]
Thanks a lot.

Yes, I decided to not put examples for the last functions, since they are in my opinion very straightforward so that the example included for the first one can simply be re-used and applied to the other functions.

I think not putting examples helps keep things cleaner, and easier to read through.

**Comment by FilippoOggionni** [comment]
Until now, the "policy" was to provide an example for each function, even if it was straightforward, but I see your reasoning and what you say makes sense. This is something we can discuss on Friday, so that we can find a shared "guideline" (another similar case is the [dependent variable](https://tudatpy.readthedocs.io/en/latest/dependent_variable.html) module). 

I have also tested all the functions with examples and they all work fine on my local build. For now I am going to close the issue and move it to the review, in case @DominicDirkx wants to have a look. Otherwise, Dominic, could you please move it to "done"? 🙂

**Comment by DominicDirkx** [comment]
This looks very good, the one thing I would add is a reference, as requested. For time scales, I find chapter 2 of this document:

https://www.usno.navy.mil/USNO/astronomical-applications/publications/Circular_179.pdf/view

very clear.

Concerning the versioning, it's on the agenda :)

**Comment by gaffarelj** [comment]
Perfect. I will add this reference, then move it to "done".

**Comment by gaffarelj** [comment]
The new reference has been added as part of commit [3841031](https://github.com/tudat-team/tudat-multidoc/commit/3841031635436959f03d2fa37f2fa4ffcea15a73).

**Comment by gaffarelj** [comment]
I reopened this issue so I don't forget to proof-read the API documentation page one more time. Some placeholders are still on it.

**Comment by gaffarelj** [comment]
Fixed with [c919846](https://github.com/tudat-team/tudat-multidoc/commit/c91984667d36ca49c4a155cc30593b899b98ea4c).
